### ISSUE TITLE
[BUG] Empty rulesets array passes silently with misleading message

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#159]
## Description

When `[extends]` is configured with a registry but `rulesets = []` is empty, tier validation passes with a misleading message saying "No extends configured" when extends IS configured - it just has no rulesets.

## Steps to Reproduce

1. Create `check.toml`:
```toml
[extends]
registry = "github:example/standards"
rulesets = []
```

2. Create `repo-metadata.yaml`:
```yaml
tier: production
```

3. Run:
```bash
cm validate tier
```

## Expected Result

A warning that the configuration is effectively useless:
```
⚠ Warning: [extends] is configured but rulesets is empty - no standards will be inherited

✓ Tier validation passed
  Tier: production (source: repo-metadata.yaml)
  Rulesets: [] (empty - no tier constraint applied)
```

## Actual Result

```
✓ Tier validation passed
  Tier: production (source: repo-metadata.yaml)
  No extends configured (no tier constraint)
```

## Problems

1. **Misleading message**: Says "No extends configured" but `[extends]` IS configured with a registry URL
2. **Silent pass**: No warning that the configuration is effectively useless
3. **User confusion**: User may think they're inheriting org standards when they're actually not

## Impact

A user might:
1. Copy a template with `[extends]` configured
2. Accidentally clear the rulesets array
3. Think they're compliant with org standards
4. Never realize no standards are being applied

## Suggested Fix

1. Change message from "No extends configured" to "No rulesets specified (no tier constraint)"
2. Add a warning when `[extends].registry` is set but `rulesets` is empty
3. Consider making empty rulesets with a registry configured a validation error

## Environment

- check-my-toolkit version: 1.4.0
- Node.js version: 25.2.1
- Platform: macOS (Darwin 25.2.0)
